packages/react/src/use-assistant.ts

```diff
@@ -160,67 +169,50 @@ export function useAssistant({
         throw new Error('The response body is empty.');
       }
 
-      for await (const { type, value } of readDataStream(
-        result.body.getReader(),
-      )) {
-        switch (type) {
-          case 'assistant_message': {
-            setMessages(messages => [
-              ...messages,
-              {
-                id: value.id,
-                role: value.role,
-                content: value.content[0].text.value,
-              },
-            ]);
-            break;
-          }
-
-          case 'text': {
-            // text delta - add to last message:
-            setMessages(messages => {
-              const lastMessage = messages[messages.length - 1];
-              return [
-                ...messages.slice(0, messages.length - 1),
-                {
-                  id: lastMessage.id,
-                  role: lastMessage.role,
-                  content: lastMessage.content + value,
-                },
-              ];
-            });
+      for await (const { value } of readDataStream(result.body.getReader())) {
+        const { event, data } = value as AssistantStreamPart;
+        setStatus('in_progress');
+        setThreadStatus(event);
 
+        switch (event) {
+          case 'thread.run.created': {
+            setThreadId(data.threadId);
             break;
           }
 
-          case 'data_message': {
+          case 'thread.message.created': {
             setMessages(messages => [
               ...messages,
               {
-                id: value.id ?? generateId(),
-                role: 'data',
+                id: data.id,
+                role: 'assistant',
                 content: '',
-                data: value.data,
               },
             ]);
+
             break;
           }
 
-          case 'assistant_control_data': {
-            setThreadId(value.threadId);
```

Comment:
how are we setting the thread id now?

Reply:
It's set [here](https://github.com/vercel/ai/pull/1949/files#diff-1d8537a4d65dc935c7f758a49b447fee82412d5a4fdb6daf8281131c371ee1f2R190-R193) when the even type is `thread.run.created`